### PR TITLE
[fix bug 1209429] Updates to TP tour for Firefox 42

### PR DESCRIPTION
**Pull request by alexgibson**

### CODE REVIEW

**Comment by alexgibson** [comment]
@flodolo as per [the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1209259#c6), strings should be final here too.

**Comment by flodolo** [comment]
Strings extracted locally, please warn if anything changes. I plan to expose them on Friday if I don't hear any objection.

**Comment by alexgibson** [comment]
Looks like more string changes are being talked about, so hold fire.

**Comment by alexgibson** [comment]
Minor string change (a punctuation mark moved inside the quotes for step 3).

**Comment by alexgibson** [comment]
Ok this one is ready for code review - @jpetto r?

Testing instructions (please use Firefox 42 beta):

1. Make sure you have UITour set to up work locally (see bedrock docs).
2. Set `privacy.trackingprotection.introURL` to point to your local bedrock instance, e.g. "http://127.0.0.1:8000/%LOCALE%/firefox/%VERSION%/tracking-protection/start/".
3. Open a new private browsing window and click the blue button that says "See how this works".

If all has gone according to plan, the tour should open and you should see a door-hanger in the corner of the page to start the tour.

File: media/css/firefox/tracking-protection-tour.less [hunk 1]
```diff
@@ -228,20 +232,23 @@ main > header {
...
 
             &:hover,
             &:focus {
+                border-color: #5f963d;
                 background-color: #6bac44;
             }
         }
     }
 
+    // https://mxr.mozilla.org/mozilla-central/source/toolkit/themes/windows/global/icons/close@2x.png
```
Comment (jpetto) [inline media/css/firefox/tracking-protection-tour.less]: Guessing these `mxr.` comments are kept as a reference to the original source images?

File: media/js/firefox/tracking-protection-tour.js [hunk 2]
```diff
@@ -138,6 +142,19 @@ if (typeof Mozilla === 'undefined') {
...
+    /**
+     * Tempoaray workaround for resizing the window (Bug 1188400)
+     */
+    TPTour.handleResize = function() {
+        if (TPTour.state === 'step1' && TPTour.panelVisible) {
+            clearTimeout(_resizeTimeout);
+            Mozilla.UITour.hideInfo();
+            TPTour.panelVisible = false;
+            _resizeTimeout = setTimeout(TPTour.showTourStep, 300);
+        }
```
Comment (jpetto) [inline media/js/firefox/tracking-protection-tour.js]: Would it be overkill to debounce this resize handler?

**Comment by jpetto** [comment]
Just making sure there's still nothing we can do to keep the step 1 dialog from hanging off to the left:

screen shot 2015-10-20 at 11 47 17 am

**Comment by jpetto** [comment]
Looking good to me. :hamburger: A couple questions, but nothing at all major. r+, pending responses.

File: media/css/firefox/tracking-protection-tour.less (hunk 1 above)
Comment (alexgibson) [inline media/css/firefox/tracking-protection-tour.less]: Yep, if any poor soul needs to find the original assets again :ghost:

File: media/js/firefox/tracking-protection-tour.js (hunk 2 above)
Comment (alexgibson) [inline media/js/firefox/tracking-protection-tour.js]: We need the handler to be pretty snappy, but now that you mention it I think we can optimize this a bit more :+1:

**Comment by alexgibson** [comment]
> Just making sure there's still nothing we can do to keep the step 1 dialog from hanging off to the left:

Yeah there's a [tracking bug open](https://bugzilla.mozilla.org/show_bug.cgi?id=1207419) to fix all these edge-case panel oddities. Most of it is the realm of one guy :neckbeard: who is the only person who understands the layout code, so no idea how long it will take to get these things fixed.

File: media/css/firefox/tracking-protection-tour.less (hunk 1 above)
Comment (jpetto) [inline media/css/firefox/tracking-protection-tour.less]: Our future selves thank you for your consideration. :trophy:

**Comment by alexgibson** [comment]
Thanks @jpetto - updated the resize handler to be a little more optimized. If all looks, ok, I'll squash and we can merge :+1:

File: tests/unit/spec/firefox/tracking-protection-tour.js [hunk 3]
```diff
@@ -247,10 +248,35 @@ describe('tracking-protection-tour.js', function() {
...
+
+        beforeEach(function() {
+            spyOn(Mozilla.UITour, 'hideInfo');
+            spyOn(Mozilla.TPTour, 'showTourStep');
+        });
+
+        it('should hide and reshow info panel if on tour step 1', function() {
+            Mozilla.TPTour.handleResize();
+            expect(Mozilla.UITour.hideInfo).toHaveBeenCalled();
+            expect(Mozilla.TPTour.panelVisible).toBeFalsy();
```
Comment (jpetto) [inline tests/unit/spec/firefox/tracking-protection-tour.js]: Don't need this test anymore?

Comment (alexgibson) [inline tests/unit/spec/firefox/tracking-protection-tour.js]: Oh good catch :+1:

**Comment by jpetto** [comment]
The optimization is real! Resizing the window is noticeably snappier. I think one orphaned line in a test can be removed, but after that, squash and merge!

**Comment by alexgibson** [comment]
Thanks @jpetto - updated and squarshed.

**Pull request merged**